libtransmission/utils.cc

```diff
@@ -193,6 +194,14 @@ bool tr_file_save(std::string_view filename, std::string_view contents, tr_error
     {
         return false;
     }
+#ifndef _WIN32
+    // set file mode per settings umask()
+    {
+        auto const val = ::umask(0);
+        fchmod(fd, 0666 - val);
```

Comment:
Are you sure this is correct?

The idiom I usually see is `fchmod(fd, 0666 & ~val);`

Reply:
First hit on Google that I saw looking for `filetype:c "0666" "fchmod"` is a `tmpfile()` function similar to what we're doing in utils.cc, and its implementation is:

https://android.googlesource.com/platform/bionic/+/ics-mr0/libc/stdio/tmpfile.c

```c++
fd = mkstemp(buf);
if (fd != -1) {
    ...
    u = umask(0);
    (void)umask(u);
    (void)fchmod(fd, 0666 & ~u);
    ...
}
```

Reply:
Can confirm the current code does not work correctly for umask `033`. My bad for missing that and approving.

Reply:
@ckerr @tearfur Thank you both!  I've pushed a correction to use bitwise complement.  Sorry for my thinko.

I've tested this correciton with the default umask 022 (-rw-r--r--), 543 (--w--w-r--), and 330 (-r--r--rw-).